src/Halogen/Component.purs

```diff
@@ -147,7 +72,41 @@ component = mkComponent <<< go
 type ComponentSpec h s f act ps i o m =
   { initialState :: i -> s
   , render :: s -> h (ComponentSlot h ps m act) act
-  , eval :: HalogenQ f act i ~> HalogenM' s act ps o m
+  , eval :: HalogenQ f act i ~> HalogenM s act ps o m
+  }
+
+type EvalSpec s f act ps i o m =
+  { handleAction :: act -> HalogenM s act ps o m Unit
+  , handleQuery :: forall a. f a -> HalogenM s act ps o m (Maybe a)
+  , receive :: i -> Maybe act
+  , initialize :: Maybe act
+  , finalize :: Maybe act
```

Comment:
I guess these could be `Maybe (HalogenM s act ps o m Unit)` rather than going via action handling, but I kinda like handling these in one place (which is already true if `HalogenQ` is handled rather than going the `mkEval` route).

Reply:
I don't think they would need to be `Maybe`, since they can just be `pure unit`.